src/ImmersedBoundaries/immersed_boundary_condition.jl

```diff
@@ -197,8 +197,8 @@ end
     q̃ᴮ = bottom_ib_flux(i, j, k, ibg, bc.bottom, loc, c, closure, K, id, clock, fields)
     q̃ᵀ =    top_ib_flux(i, j, k, ibg, bc.top,    loc, c, closure, K, id, clock, fields)
 
-    iᵂ, jˢ, kᴮ = index_right.((i, j, k), loc)
-    iᴱ, jᴺ, kᵀ = index_left.((i, j, k), loc)
+    iᵂ, jˢ, kᴮ = map(index_right, (i, j, k), loc) # Broadcast instead of map causes inference failure
```

Comment:
💡

Reply:
broadcast is a tricky to infer functions, map is much easier

Reply:
makes sense given the complexity of broadcast